CONTRIBUTOR-DOCS/01_contributor-guides/04_making-a-pull-request.md

```diff
@@ -0,0 +1,181 @@
+
+### Branch naming
+
+We use a straightforward branch naming convention:
+
+- `[username]/[short-description]` (e.g., `alex/fix-dropdown-bug`)
+- If referencing a known issue, incorporate the issue number (e.g., `alex/123-fix-dropdown-bug`)
+
+### Changeset requirements
+
+For PRs that add or update a component:
+
+- Must include a changeset to trigger the release train and update the CHANGELOG
+- Changeset type should be one of:
+    - `patch` - for bug fixes only
+    - `minor` - for new components or new APIs in an existing component
+    - `major` - for breaking changes to a component or public library API
+
+### Conventional commits
+
+We follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) specification to make change tracking predictable:
+
+Format: `type(component?): subject`
+
+The component is optional but should reference the package you are updating.
+
+Types include:
+
+- `feat`: New features or enhancements
+- `fix`: Bug fixes
+- `docs`: Documentation changes
+- `style`: Formatting, linting (not CSS changes)
+- `chore`: Build tooling, repo management, dependency updates
+- `perf`: Performance improvements
+- `test`: Adding or updating tests
+
+Examples:
+
+- `feat(sp-card): add shadow styles for theme consistency`
+- `fix(sp-action-menu): correct arrow key navigation in nested menus`
+- `docs: clarify how to submit bug reports`
+
+For breaking changes, add a `!` after the type/scope:
+
+- `feat(sp-button)!: change API for icon placement`
+
+---
+
+## Submitting your PR
+
+### Pull request template
+
+When creating a pull request, you'll be presented with our template. Complete all sections to the best of your ability, including:
+
+- Description of the changes
+- Related issues (using proper [GitHub keywords](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests#linking-a-pull-request-to-an-issue) to auto-close issues i.e. `fixes`, `resolves`, or `closes`)
+- Type of change in the PR title (bug fix, feature, breaking change)
+- Steps you took to test your changes that reviewers can follow to also test them
+- Checklist of items completed
+- Screenshots/videos for visual changes
+
+Incomplete templates may delay the review process.
+
+### Labels for PR authors
+
+As a PR author, you can use these labels to communicate the status of your pull request:
+
+- `ready-for-review`: PR is ready for maintainer review
+- `WIP`: PR is still being worked on, not ready for review
+- `needs-self-review`: You plan to do a self-review before requesting maintainer review
+- `help-wanted`: You need extra attention or assistance on this PR
+- `breaking-change`: PR contains changes that break backward compatibility
+- `Component: [Name]`: PR affects this component
+
+For a complete list of labels and their meanings, including reviewer-specific labels, see [Participating in PR reviews](05_participating-in-pr-reviews.md#labels-and-their-meanings).
```

Comment:
I kind of don't want to have this live in a markdown file--I'd rather link to our github labels directly just because it's a lot of overhead to maintain.